cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorTest.java

```diff
@@ -766,6 +769,18 @@ public void testUpdateClusterNoop(Params params, VertxTestContext context) {
         updateCluster(context, getKafkaAssembly("bar"), kafkaAssembly);
     }
 
+    @ParameterizedTest
+    @MethodSource("data")
+    public void testUpdateClusterKafkaVersion(Params params, VertxTestContext context) {
+        setFields(params);
+        Kafka kafkaAssembly = getKafkaAssembly("bar");
+        kafkaAssembly.setStatus(new KafkaStatusBuilder()
+                .withKafkaVersion("3.4.0")
+                .withOperatorLastSuccessfulVersion("0.30.0")
+                .build());
+        updateCluster(context, getKafkaAssembly("bar"), kafkaAssembly);
+    }
```

Comment:
Does this really need to be in the parametrized tests? This part seems to have nothing with the paramettrization. So maybe it would fit better in the non-paramterized tests?